core/src/edu/unibo/martyadventure/view/MovementGameScreen.java

```diff
@@ -1,58 +1,194 @@
+        public static float viewportHeight;
+        public static float virtualWidth;
+        public static float virtualHeight;
+        public static float physicalWidth;
+        public static float physicalHeight;
+        public static float aspectRatio;
+}
+    
     private PlayerCharacterView player;
+    private EnemyCharacterView biff;
     private PlayerInputProcessor inputProcessor;
-    private TextureRegion currentFrame;
-    private Sprite currentSprite;
+    private TextureRegion playerCurrentFrame;
+    private TextureRegion biffCurrentFrame;
     private OrthogonalTiledMapRenderer mapRenderer;
     private OrthographicCamera camera;
     private static MapManager mapManager;
     
-    public MovementGameScreen() {
+    
+    public MovementGameScreen(){
         mapManager = new MapManager();
     }
 
+    /**
+     * Setup the screen elements
+     */
     @Override
     public void show() {
         //camera
+        setupViewport(50, 50);
         camera = new OrthographicCamera();
-        camera.setToOrtho(false, 10 , 10);
+        camera.setToOrtho(false, VIEWPORT.viewportWidth, VIEWPORT.viewportHeight);
         
         //rederer
+        try {
+            mapManager.loadMap(MapManager.Maps.MAP1);
+        } catch (InterruptedException | ExecutionException | IOException e) {
+            e.printStackTrace();
+        }
+        
+       
         mapRenderer = new OrthogonalTiledMapRenderer(mapManager.getCurrentMap(), MapManager.UNIT_SCALE);
         mapRenderer.setView(camera);
         
-        System.err.println("Unit scale:" + MapManager.UNIT_SCALE);
+        //player
+        try {
+            player = new PlayerCharacterView(mapManager.getPlayerStartPosition());
+        } catch (InterruptedException | ExecutionException e) {
+            e.printStackTrace();
+        }
         
         //player
+        try {
+            biff = new EnemyCharacterView(mapManager.getBiffStartPosition());
+            biff.setDirection(EntityDirection.DOWN);
+        } catch (InterruptedException | ExecutionException e) {
+            e.printStackTrace();
+        }
         
+        inputProcessor = PlayerInputProcessor.getPlayerInputProcessor();
+        inputProcessor.setPlayer(player);
+        Gdx.input.setInputProcessor(inputProcessor);
         
 
     }
 
+    /**
+     * Called every frame, update the current screen view
+     */
     @Override
     public void render(float delta) {
-        // TODO Auto-generated method stub
+        Gdx.gl.glClearColor(0, 0, 0, 1);
+        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
+        
+        //set the camera position
+        camera.position.set(player.getCurrentPosition().x,player.getCurrentPosition().y,0f);
+        camera.update();
+        //update the current player frame
+        playerCurrentFrame = player.getCurrentFrame();
+        biffCurrentFrame = biff.getCurrentFrame();
+        
+        //check collisions
+        try {
```

Comment:
Stesso discorso di prima: normalmente in Java se qualcosa si rompe, esci con errore invece di cercare di tirare avanti

Reply:
più avanti chiamerò un metodo dello screen manager che segnalerà un errore nel gioco e chiuderà il gioco